src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,73 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
 
+    public static final String SUPPLY = "supply";
+    public static final String COMMA = ",";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fileRead = new File(fromFileName);
+        File fileWrite = new File(toFileName);
+        StringBuilder data = new StringBuilder();
+        StringBuilder reportData = new StringBuilder();
+        fileWrite.getAbsoluteFile().delete();
+        writeToStringBuilder(fileRead, data);
+        String[] dataArray = data.toString().split(System.lineSeparator());
+        reportData.append(SUPPLY)
+                .append(COMMA)
+                .append(getTotalAmount(dataArray, SUPPLY))
```

Comment:
we call this method a lot of times, let's do it one time and return. 
for example, we can move logic from method getTotalAmount() to the method getStatistic() 
and we can create separate private method for append the result. It can be `String createStatistic(int supply, int buy)`